lib/iris/cube.py

```diff
@@ -2168,23 +2161,14 @@ def new_cell_measure_dims(cm_):
                     for d in self.cell_measure_dims(cm_)
                     if dimension_mapping[d] is not None]
 
-        try:
-            first_slice = next(slice_gen)
-        except StopIteration:
-            first_slice = None
-
+        # Fetch the data as a generic array-like object.
         cube_data = self._data_manager.core_data
 
-        if first_slice is not None:
-            data = cube_data[first_slice]
-        else:
-            data = copy.deepcopy(cube_data)
-
-        for other_slice in slice_gen:
-            data = data[other_slice]
+        # Index with the keys, using orthogonal slicing.
+        dimension_mapping, data = iris.util._slice_data_with_keys(
```

Comment:
It is used as a free variable in the two methods defined immediately _above_, `new_coord_dims` and `new_cell_measure_dims`.
Not obvious, is it ?!?
I considered moving them to below this point for more clarity, but the minimal change principle won out...